test/unit/query-generator/select-query.test.ts

```diff
@@ -7,6 +7,10 @@ import { createSequelizeInstance, expectsql, sequelize } from '../../support';
 describe('QueryGenerator#selectQuery', () => {
   const queryGenerator = sequelize.getQueryInterface().queryGenerator;
 
+  if (sequelize.dialect.name === 'mssql') {
+    sequelize.options.databaseVersion = '11.0.0';
```

Comment:
I don't like that we modify `databaseVersion` here. The same sequelize instance is used for all unit tests and modifying the instance will impact the test results.

Instead consider creating a new sequelize instance using `createSequelizeInstance`. You can pass `databaseVersion` to its options.

If the test relies on databaseVersion, it would be a good idea to add a test that checks what the output is like with another databaseVersion (e.g. if the code checks whether the version is >= 11, then we should test against databaseVersion 10 & 11).

Note that we're dropping mssql < 11 in v7 (https://github.com/sequelize/sequelize/pull/15249). Let's merge that other PR so we can remove these version assignments altogether

Reply:
With that PR merged I think we can undo all the changes of this PR to the unit tests